pkg/controllers/csi/driver/server.go

```diff
@@ -233,25 +238,37 @@ func (svr *Server) NodeExpandVolume(context.Context, *csi.NodeExpandVolumeReques
 
 func logGRPC() grpc.UnaryServerInterceptor {
 	return func(ctx context.Context, req any, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (any, error) {
-		if info.FullMethod == "/csi.v1.Identity/Probe" || info.FullMethod == "/csi.v1.Node/NodeGetCapabilities" {
-			return handler(ctx, req)
-		}
+		var methodName string
 
-		methodName := ""
-
-		if info.FullMethod == "/csi.v1.Node/NodePublishVolume" {
+		switch {
+		case info.FullMethod == "/csi.v1.Node/NodePublishVolume":
 			req := req.(*csi.NodePublishVolumeRequest)
 			methodName = "NodePublishVolume"
 			log.Info("GRPC call", "method", methodName, "volume-id", req.GetVolumeId())
-		} else if info.FullMethod == "/csi.v1.Node/NodeUnpublishVolume" {
+		case info.FullMethod == "/csi.v1.Node/NodeUnpublishVolume":
 			req := req.(*csi.NodeUnpublishVolumeRequest)
 			methodName = "NodeUnpublishVolume"
 			log.Info("GRPC call", "method", methodName, "volume-id", req.GetVolumeId())
+		default:
+			resp, err := handler(ctx, req)
+			if err != nil {
+				log.Error(err, "GRPC failed", "full_method", info.FullMethod)
+			}
+
+			return resp, err
+		}
+
+		if counter.Load() > MaxGrpcRequests {
```

Comment:
Like the rework into a switch statement. 

Would rather implement the rate limiter as an own interceptor though, it's not really related to `logGRPC`.

Reply:
I think just renaming it would make sense, the "log" interceptor is kinda redundant at that point.
We are rate limiting, and log as a side thing

Reply:
For you to have 2 interceptors, you would (AFAIK) need an `grpc_middleware.WithUnaryServerChain` for it to be called in order. And that would require a new dependency AND just overcomplicate things

Reply:
just call it `grpcLimiter` or something like that instead of `logGRPC`

Reply:
agree with @0sewa0 about renaming.

Reply:
the problem that we don't have another logic :) except to log and limit, we want log when we limit as well. So create 2 functions sounds overkill.